src/evo/deterministicmns.cpp

```diff
@@ -885,6 +885,7 @@ bool CDeterministicMNManager::BuildNewListFromBlock(const CBlock& block, gsl::no
             }
 
             auto newState = std::make_shared<CDeterministicMNState>(*dmn->pdmnState);
+            newState->nVersion = opt_proTx->nVersion;
```

Comment:
So far as I see, a newState here is an internal data structure; it should not have version.
Version should be used _only_ for serialization / deserialization / send by network.

Could you justify nVersion usage here, please?

Reply:
Follows precedent set by `pubKeyOperator` for consistency's sake (setting `nVersion` when (re)setting any field that is version-dependent)

https://github.com/dashpay/dash/blob/5cbddbb17f660b7e4c183d427aa2d567745d0541/src/evo/deterministicmns.cpp#L940-L943

Reply:
Wrapped it in `v23_active` ([diff](https://github.com/dashpay/dash/compare/d26fde0f478313924efa63310e6d24e135d49de6..64f3219b985d67648190a24818cc94b3d7181329#diff-0998f8dfc4c1089e90cbaafe9607b361035b904cd103df31e3c2339a3cbf790dR839-R842)), resolved in latest push.